packages/web-components/fast-element/src/components/element-controller.ts

```diff
@@ -180,22 +190,19 @@ export class ElementController<TElement extends HTMLElement = HTMLElement>
             }
         }
 
-        // Capture any observable values that were set by the binding engine before
-        // the browser upgraded the element. Then delete the property since it will
-        // shadow the getter/setter that is required to make the observable operate.
-        // Later, in the connect callback, we'll re-apply the values.
+        // Delete and re-assign any observable properties that were assigned
+        // before the element is upgraded. This needs to happen because otherwise
+        // the value will shadow the observable getter and setter
         const accessors = Observable.getAccessors(element);
 
         if (accessors.length > 0) {
-            const boundObservables = (this.boundObservables = Object.create(null));
-
             for (let i = 0, ii = accessors.length; i < ii; ++i) {
-                const propertyName = accessors[i].name;
+                const propertyName = accessors[i].name as keyof TElement;
                 const value = (element as any)[propertyName];
 
                 if (value !== void 0) {
-                    delete (element as any)[propertyName];
-                    boundObservables[propertyName] = value;
+                    delete element[propertyName];
+                    element[propertyName] = value;
```

Comment:
Is this going to cause a regression? I know we have to delete and reset....I thought it needed to be done in the connectedCallback. Could this cause an error to be thrown? For example, if setting a property then reflects to an attributes...I think that's forbidden. This might need to have changes in the attribute implementation to go along with it.

Reply:
I realized after I did this work it doesn't need to happen right now; I originally thought I needed to run the re-binding in both classes, so instead of adding a method I just re-assigned the value in the ctor.

It might be worth considering now though, as a breaking change. There is an existing odd behavior where the value *can* be different between constructor and connect calls: https://stackblitz.com/edit/typescript-d5xlyg?file=index.ts. We would definitely need to update the attribute reflection behavior to defer reflection.

Reply:
I'm open to making the change.

Reply:
I'm going to investigate this separately; it doesn't need to happen in this PR.